site/docs/develop/spec.md

```diff
@@ -1,80 +1,180 @@
+The state of a Delta Lake Table is represented using the Delta Lake Table attributes
+`metadataLocationHistory`, `checkpointLocationHistory` and `lastCheckpoint`.
 
-## Tiered Version Store
+Delta Lake Tables are tracked without a _Global State_ in Nessie, i.e. those three attributes are
+recorded within the [_Put Operation_](#put-operation) of a Nessie commit.
+
+#### View
+
+The state of an SQL view is represented using the attributes
+`sqlText` and `dialect` (currently one of `HIVE`, `SPARK`, `DREMIO`, `PRESTO`).
+
+SQL views are tracked without a _Global State_ in Nessie, i.e. those three attributes are
+recorded within the [_Put Operation_](#put-operation) of a Nessie commit.
+
+## Operations in a Nessie commit
+
+Each Nessie commit carries one or more operations. Each operation contains the Content Key plus and
+comes in one of the following variations.
+
+### Put operation
+
+A _Put operation_ modifies the state of the included Content object. It must contain the Content
+object and, if the Content type tracks [_Global State_](#on-reference-state-vs-global-state), also
+the _expected contents_. The _expected contents_ attribute can be omitted, if the Content object
+refers to a new Content Id, e.g. a newly created table or view.
+
+### Delete operation
+
+A _Delete operation_ does not carry any Content object and is used to indicate that a Content
+object is no longer referenced using the Content Key of the _Delete operation_.
+
+### Unmodified operation
+
+An _Unmodified operation_ does not represent any change of the data, but can be included in a
+Nessie commit operation to enforce strict serializable transactions. The presence of an
+_Unmodified operation_ means that the Content object referred to via the operation's Content Key
+must not have been modified since the Nessie commit's `expectedHash`.
+
+## Version Store
+
+See [Commit Kernel](kernel.md) for details.
+
+### Conflict Resolution
 
-!!! warning
-    todo
+The API passes an `expectedHash` parameter with a Nessie commit operation. This is the commit that
+the client thinks is the most up to date (its HEAD). The Nessie backend will check to see if the
+key has been modified since that `expectedHash` and if, it will reject the requested modification
+with a `NessieConflictException`. This is basically an optimistic lock that accounts for the fact
+that the commit hash is global and nessie branch could have moved on from `expectedHash` without
+modifying the key in question.
+
+For content tables that require Global State, a Nessie [_Put operation_](#put-operation) should
+pass the so called _expected state_, which will be used to compare the recorded Global State of a
+content object with the Global State in the _expected state_ in the _Put operation_. If both values
+differ, Nessie will reject the operation with a `NessieConflictException`.
```

Comment:
Is it good to add what need to be done to merge that commit successfully after receiving  `NessieConflictException` ? Like do we need to rebase or need to cherry-pick commits manually instead of  merge ?

Reply:
It's not really part of the Nessie spec. A conflict needs to be resolved on a case-by-case basis. Currently the user has to re-do all changes manually. But we want to have at least Iceberg itself help us with that, because the Iceberg code can already handle certain types of conflicts: e.g. an APPEND on the target branch which currently causes a conflict can be easily resolved by Iceberg. But all this requires a lot more thoughts.

Reply:
ok. We can update once we have clean guidelines.